src/app/components/team/Languages/LanguagesComponent.js

```diff
@@ -1,66 +1,148 @@
-/* eslint-disable @calm/react-intl/missing-attribute, relay/unused-fields */
 import React from 'react';
 import { PropTypes } from 'prop-types';
-import { FormattedMessage } from 'react-intl';
-import { createFragmentContainer, graphql } from 'react-relay/compat';
-import Card from '@material-ui/core/Card';
-import CardContent from '@material-ui/core/CardContent';
+import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
+import { createFragmentContainer, graphql, commitMutation } from 'react-relay/compat';
+import Relay from 'react-relay/classic';
 import List from '@material-ui/core/List';
 import AddLanguageAction from './AddLanguageAction';
 import LanguageListItem from './LanguageListItem';
 import SettingsHeader from '../SettingsHeader';
-import { safelyParseJSON } from '../../../helpers';
-import { compareLanguages } from '../../../LanguageRegistry';
+import { FlashMessageSetterContext } from '../../FlashMessage';
+import GenericUnknownErrorMessage from '../../GenericUnknownErrorMessage';
+import SwitchComponent from '../../cds/inputs/SwitchComponent';
+import { safelyParseJSON, getErrorMessageForRelayModernProblem } from '../../../helpers';
+import { compareLanguages, languageLabelFull } from '../../../LanguageRegistry';
 import { ContentColumn } from '../../../styles/js/shared';
+import styles from './LanguagesComponent.module.css';
+
+const submitToggleLanguageDetection = ({
+  team,
+  value,
+  onSuccess,
+  onFailure,
+}) => {
+  commitMutation(Relay.Store, {
+    mutation: graphql`
+      mutation LanguagesComponentToggleLanguageDetectionMutation($input: UpdateTeamInput!) {
+        updateTeam(input: $input) {
+          team {
+            id
+            get_language_detection
+          }
+        }
+      }
+    `,
+    variables: {
+      input: {
+        id: team.id,
+        language_detection: value,
+      },
+    },
+    onError: onFailure,
+    onCompleted: ({ data, errors }) => {
+      if (errors) {
+        return onFailure(errors);
+      }
+      return onSuccess(data);
+    },
+  });
+};
 
 const LanguagesComponent = ({ team }) => {
   const defaultCode = team.get_language || 'en';
+  const setFlashMessage = React.useContext(FlashMessageSetterContext);
 
   let languages = safelyParseJSON(team.get_languages) || [];
   languages = languages.sort((a, b) => compareLanguages(defaultCode, a, b));
 
+  const toggleLanguageDetection = (value) => {
+    const onFailure = (errors) => {
+      setFlashMessage((
+        getErrorMessageForRelayModernProblem(errors)
+        || <GenericUnknownErrorMessage />
+      ), 'error');
+    };
+
+    submitToggleLanguageDetection({
+      team,
+      value,
+      onSuccess: () => {},
```

Comment:
I see the design has nothing happening on success here, so I guess this is not a big problem but want to flag for @brianfleming that we probably want to figure out some kind of generic "success" indicator for all our invisible inputs where you flip a switch and we quickly talk to check-api to toggle something.